crates/rpcsrv/src/flight/handler.rs

```diff
@@ -140,11 +100,68 @@ impl FlightSessionHandler {
             .map_err(Status::from);
         Ok(Response::new(Box::pin(stream)))
     }
+
+    pub fn new(engine: &Arc<Engine>) -> Self {
+        Self {
+            engine: engine.clone(),
```

Comment:
```suggestion
    pub fn new(engine: Arc<Engine>) -> Self {
        Self {
            engine,
```

Just a minor nitpick. I prefer passing in owned values (if required) inside the function arguments. It avoids the `clone`.